Modules/Loadable/Segmentations/EditorEffects/qSlicerSegmentEditorAbstractLabelEffect.cxx

```diff
@@ -120,13 +120,25 @@ void qSlicerSegmentEditorAbstractLabelEffect::updateMRMLFromGUI()
 }
 
 //-----------------------------------------------------------------------------
-void qSlicerSegmentEditorAbstractLabelEffect::appendPolyMask(vtkOrientedImageData* input, vtkPolyData* polyData, qMRMLSliceWidget* sliceWidget)
+void qSlicerSegmentEditorAbstractLabelEffect::appendPolyMask(vtkOrientedImageData* input, vtkPolyData* polyData, qMRMLSliceWidget* sliceWidget, vtkMRMLSegmentationNode* segmentationNode/*=nullptr*/)
 {
   // Rasterize a poly data onto the input image into the slice view
   // - Points are specified in current XY space
   vtkSmartPointer<vtkOrientedImageData> polyMaskImage = vtkSmartPointer<vtkOrientedImageData>::New();
   qSlicerSegmentEditorAbstractLabelEffect::createMaskImageFromPolyData(polyData, polyMaskImage, sliceWidget);
 
+  // If the segmentation has a parent transform, apply a transformation to polyMaskImage to move it to the same coordinate system.
+  if (segmentationNode && segmentationNode->GetParentTransformNode())
+    {
+    vtkNew<vtkMatrix4x4> worldToSegmentation;
+    segmentationNode->GetParentTransformNode()->GetMatrixTransformFromWorld(worldToSegmentation);
```

Comment:
Check if transform to world is linear (transformNode->IsTransformToWorldLinear()) and if not then log a warning message and ignore the transform (this is what we do in qSlicerSegmentEditorAbstractLabelEffect::imageToWorldMatrix).

Reply:
Done!